src/js/components/ToggleButtonGroup/ToggleButtonGroup.js

```diff
@@ -0,0 +1,182 @@
+      );
+    }
+    return defaultIndex;
+  }, [value, values]);
+
+  const [focusableIndex, setFocusableIndex] = useState(() =>
+    getFocusableIndex(),
+  );
+
+  const onNext = (e) => {
+    // prevent page scroll
+    e.preventDefault();
+    const nextIndex =
+      focusableIndex + 1 <= options.length - 1 ? focusableIndex + 1 : 0;
+    setFocusableIndex(nextIndex);
+    buttonRefs.current[nextIndex].focus();
+  };
+
+  const onPrevious = (e) => {
+    // prevent page scroll
+    e.preventDefault();
+    const nextIndex =
+      focusableIndex - 1 >= 0 ? focusableIndex - 1 : options.length - 1;
+    setFocusableIndex(nextIndex);
+    buttonRefs.current[nextIndex].focus();
+  };
+
+  const handleToggle = (option, active) => {
+    let nextValue;
+    if (multiple) {
+      if (Array.isArray(value)) {
+        nextValue = active
+          ? value.filter((item) => item !== option)
+          : [...value, option];
+      } else {
+        nextValue = active ? [] : [option];
+      }
+    } else {
+      nextValue = active ? '' : option;
+    }
+    setValue(nextValue);
+  };
+
+  return (
+    <Keyboard
+      onUp={onPrevious}
+      onDown={onNext}
+      onLeft={onPrevious}
+      onRight={onNext}
+    >
+      <Box
+        ref={ref}
+        direction="row"
+        alignSelf="start"
+        role="group"
+        onBlur={(e) => {
+          if (!ref?.current.contains(e.relatedTarget)) {
+            setFocusableIndex(getFocusableIndex());
+          }
+        }}
+        {...theme.toggleButtonGroup.container}
+        {...rest}
+      >
+        {options.map((option, index) => {
+          let label;
+          let icon;
+          let optionValue;
+          if (typeof option === 'object') {
+            icon = option.icon;
+            label = option.label;
+            optionValue = option.value;
+          } else {
+            label = option;
+            optionValue = option;
+          }
+          const active = Array.isArray(value)
+            ? !!value.includes(optionValue)
+            : value === optionValue;
+          let round = 0;
+          // round corners of first and last buttons to match container
+          if (
+            typeof theme.toggleButtonGroup.container.round === 'string' &&
+            (index === 0 || index === options.length - 1)
+          ) {
+            round = {
+              corner: index === 0 ? 'left' : 'right',
+              size: theme.toggleButtonGroup.container.round,
+            };
+          }
+
+          return (
+            <Box
+              border={
+                index < options.length - 1
+                  ? {
+                      side: 'right',
+                      color: theme.toggleButtonGroup.divider.color,
+                    }
+                  : undefined
+              }
+              key={optionValue || index}
+            >
+              <StyledButton
+                active={active}
+                aria-checked={active}
```

Comment:
I believe we will need to choose `aria-checked` when in radio mode and `aria-pressed` when in multiple mode. Aria-pressed will announce the control as a "toggle button."

[reference](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-pressed)